var/spack/repos/builtin/packages/chapel/package.py

```diff
@@ -3,24 +3,811 @@
+    variant("chpldoc", default=False, description="Build chpldoc in addition to chpl")
+
+    variant(
+        "comm",
+        default="none",
+        description="Build Chapel with multi-locale support",
+        values=("gasnet", "none", "ofi", "ugni"),
+    )
+
+    variant(
+        "comm_substrate",
+        default="unset",
+        description="Build Chapel with GASNet multi-locale support using the "
+        "supplied CHPL_COMM_SUBSTRATE",
+        values=("ibv", "ofi", "udp", "unset"),
+        multi=False,
+        sticky=True,  # never allow the concretizer to choose this
+    )
+
+    # Chapel depends on GASNet whenever comm=gasnet.
+    # The default (and recommendation) is to use the embedded copy of GASNet.
+    # This variant allows overriding with a particular version of GASNet sources,
+    # although this is not officially supported and some combinations might be rejected.
+    variant(
+        "gasnet",
+        description="Control the GASNet library version used",
+        default="bundled",
+        values=("bundled", "spack"),
+        multi=False,
+    )
+
+    variant(
+        "gasnet_segment",
+        default="unset",
+        description="Build Chapel with multi-locale support using the "
+        "supplied CHPL_GASNET_SEGMENT",
+        values=("everything", "fast", "large", "unset"),
+        multi=False,
+    )
+
+    variant(
+        "gmp",
+        description="Build with gmp support",
+        default="spack",
+        values=("bundled", "none", "spack", "system"),
+        multi=False,
+    )
+
+    variant(
+        "gpu_mem_strategy",
+        description="The memory allocation strategy for GPU data",
+        values=("array_on_device", "unified_memory"),
+        default="array_on_device",
+        multi=False,
+    )
+
+    variant(
+        "host_arch",
+        description="Host architecture of the build machine",
+        values=("x86_64", "aarch64", "arm64", "unset"),
+        default="unset",
+        multi=False,
+    )
+
+    # Feedback that it's hard to imagine any circumstance where host_compiler or
+    # target_compiler could meaningfully differ from what Spack already knows about
+    # variant(
+    #     "host_compiler",
+    #     values=compilers,
+    #     description="Compiler suite for building the Chapel compiler on CHPL_HOST_PLATFORM",
+    #     default="unset",
+    # )
+
+    variant(
+        "host_jemalloc",
+        values=("bundled", "none", "system", "unset"),
+        default="unset",
+        multi=False,
+        description="Selects between no jemalloc, bundled jemalloc, or system jemalloc",
+    )
+
+    variant(
+        "host_mem",
+        values=("cstdlib", "jemalloc"),
+        default="jemalloc",
+        description="Memory management layer for the chpl compiler",
+        multi=False,
+    )
+
+    variant(
+        "host_platform",
+        description="Host platform",
+        default="unset",
+        values=platform_opts,
+        multi=False,
+    )
```

Comment:
> the user has an option to choose a host_platform and we'll use it if they do

(minor) I might be confused, because I don't see any logic here that uses any value other than `unset`, which is to say the following variant definition would be equivalent:

```
    variant(
        "host_platform",
        description="Host platform",
        default="unset",
        values=("set", "unset"),  # no need to define a number of host platforms here
        multi=False,
    )
```